accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,298 @@
+
+    Example:
+
+    ```
+    bootstrap_first_master:
+      cmd.run:
+        - name: skuba node bootstrap --user sles --sudo --target <node IP> <node name>
+    
+    bootstrap_additional_master:
+      cmd.run:
+        - name: skuba node join --role master --user sles --sudo --target <node IP> <node name>
+    
+    bootstrap_worker:
+      cmd.run:
+        - name: skuba node join --role worker --user sles --sudo --target <node IP> <node name>
+    ```
+
+    The output of the `skuba` command must be parsed and shown to the user in SUSE Manager in case of errors.
+
+## Manage a CaaS Platform cluster using SUSE Manager
+
+### Show cluster status
+
+SUSE Manager shows the status of the cluster by invoking `skuba cluster status` on the Management Node (via Salt).
+
+E.g. of the output:
+
+```
+NAME         OS-IMAGE                              KERNEL-VERSION        CONTAINER-RUNTIME   HAS-UPDATES   HAS-DISRUPTIVE-UPDATES
+master0   SUSE Linux Enterprise Server 15 SP1   4.12.14-110-default   cri-o://1.13.3      <none> <none>
+worker0   SUSE Linux Enterprise Server 15 SP1   4.12.14-110-default   cri-o://1.13.3      <none> <none>
+```
+
+The output is presented in the "CaaS Platform > cluster name" section.
+
+### Removing nodes from the cluster
+
+A node can also be removed from the cluster:
+
+- temporarily (cordoned off): using `kubectl` from the Management Node via Salt.
+- permanently: by invoking `skuba node remove <nodename>` via Salt
+
+Example state:
+
+```
+remove_node:
+      cmd.run:
+        - name: skuba node remove <nodename>
+```
+
+NOTE: Other `skuba` options can be implemented in the future (e.g. resetting and reconfiguring a node).
+
+### Cluster upgrade
+
+All the cluster updates must be handled by `skuba`.
+
+SUSE Manager can check and notify the user if an update for the cluster is present by triggering `skuba cluster upgrade plan` on the Management Node (using Salt) and check if a new output is available in a cron fashion (e.g. Taskomatic job).
+
+SUSE Manager will then interface with the Management node and upgrade every master in the cluster first:
+
+`skuba node upgrade apply --target <master-node-ip> --user <user> --sudo`
+
+and then every worker:
+
+`skuba node upgrade apply --target <master-node-ip> --user <user> --sudo`
+
+This can be done by supplying the command to the Management Node via Salt.
+
+### CVE Auditing
+
+By leveraging the existing feature of CVE Auditing, SUSE Manager can detect and notify the user whether the cluster is running a vulnerable image. When SUSE Manager rebuilds the image and patches it, SUSE Manager can also guide the user in modifying the Kubernetes deployment to pull the updated image (in case `imagePullPolicy:` is not `Always`) or do a `kubectl rolling-update`.
```

Comment:
I think it's really interesting that this could be a great way in which SUMA and CaaSP can benefit from each other. CaaSP if enabled the integration with SUMA, could have an admission controller that requests SUMA to confirm if a given image has any CVE, rejecting the pod creation if that's the case. For another RFC :)

Reply:
We should also consider moving the existing [Kubernetes image audit feature](https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00035-kubernetes-integration.md) from "Virtual Host Gatherers" and couple it with the new structure. This would probably too much for this RFC, but we can think about it as a follow-up.

Reply:
Added suggestions under Future work